src/uwtools/tests/test_rocoto.py

```diff
@@ -129,6 +129,25 @@ def test__add_task_dependency_fail(self, instance, root):
         with raises(UWConfigError):
             instance._add_task_dependency(e=root, config=config)
 
+    @pytest.mark.parametrize("config", [{"datadep": {"age": "120"}}, {"timedep": {"offset": "&DEADLINE;"}}])
+    @pytest.mark.parametrize("validity_tag", ["datadep", "timedep"])
+    @pytest.mark.parametrize("validity_items", [("age", "120"), ("offset", "&DEADLINE;")])
+    def test__add_operand_dependency(self, instance, root, config, validity_tag, validity_items):
+        # config = {"datadep": {"age": "120"}}
+        #breakpoint()
+        validity_key, validity_val = validity_items
+        instance._add_task_dependency_operand(e=root, config=config)
+        dependency = root[0]
+        assert dependency.tag == validity_tag
+        assert dependency.get(validity_key) == validity_val
+
+    def test__add_time_dependency(self, instance, root):
+        config = {"timedep": {"offset": "&DEADLINE;"}}
+        instance._add_task_dependency_operand(e=root, config=config)
+        timedep = root[0]
+        assert timedep.tag == "timedep"
+        assert timedep.get("offset") == "&DEADLINE;"
+
```

Comment:
Here's my first attempt of combining these two test cases by using parametrization. Quite a bit going on and still working out the logistics of it. Thought I'd share to confirm the direction we're going in.

Reply:
Stacking the decorators like that results in the Cartesian product of all those values. Since there are 2 items in each decorator, we'd get 2 x 2 x 2 = 8 different tests, one in which (for example), `config` will be `{"datadep": {"age": "120"}}`, `validity_tag` will be `timedep`, and `validity_items` will be `("age", "120")`, which I think would not be a combination.

What about something like
```
    @pytest.mark.parametrize(
        "config", [{"datadep": {"age": "120"}}, {"timedep": {"offset": "&DEADLINE;"}}]
    )
    def test__add_task_dependency_operand(self, config, instance, root):
        instance._add_task_dependency_operand(e=root, config=config)
        element = root[0]
        tag = list(config.keys())[0]
        assert element.tag == tag
        for attr, value in config[tag].items():
            assert element.get(attr) == value
```
? That implies just two tests.

Reply:
The `for` loop in my suggestion may not be even needed -- but it'd be useful if there were multiple attributes to test and would work even if two tags had different numbers of attributes.

Reply:
Here's one step cleaner, I think:
```
    @pytest.mark.parametrize(
        "config", [{"datadep": {"age": "120"}}, {"timedep": {"offset": "&DEADLINE;"}}]
    )
    def test__add_task_dependency_operand(self, config, instance, root):
        instance._add_task_dependency_operand(e=root, config=config)
        element = root[0]
        for tag, attrs in config.items():
            assert element.tag == tag
            for attr, value in attrs.items():
                assert element.get(attr) == value
```

Reply:
A couple questions/design choices here:
1. The ordering of the input parameters here should be lexicographical. 
i.e. ```def test__add_task_dependency_operand(self, config, instance, root)```

2. The ordering of the test cases here is not lexicographical and this is because it follows the ordering of how the tags appear in the YAML/workflow. I left this consistent by simply adding the new test case below this test case: ```test__add_task_dependency```

3. I also left ```taskdep`` within the ```test__add_task_dependency``` test case and not factoring it out into the ```def test__add_task_dependency_operand```. I felt like within this case it looked fine staying within the current one. And looks cleaner altogether.

Reply:
1. Since it's arbitrary which order the task arguments appear (except for `self` for test methods rather than standalone functions), I try to default to lexicographical ordering.
2. Fair enough -- there are often good reasons to depart from convention. [PEP8 agrees.](https://peps.python.org/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds).